gensim/api/__init__.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import print_function
+from __future__ import absolute_import
+import json
+import os
+import tarfile
+import logging
+import sys
+import errno
+try:
+    import urllib.request as urllib
+except ImportError:
+    import urllib
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+user_dir = os.path.expanduser('~')
+base_dir = os.path.join(user_dir, 'gensim-data')
+log_file_dir = os.path.join(base_dir, 'api.log')
+if not os.path.isdir(base_dir):
+    try:
+        os.makedirs(base_dir)
+    except OSError as e:
+        if e.errno == errno.EEXIST:
+            raise Exception(
+                "Not able to create folder gensim-data in {}. File gensim-data "
+                "exists in the direcory already.".format(user_dir))
+        else:
+            raise Exception(
+                "Can't create {}. Make sure you have the read/write permissions "
+                "to the directory or you can try creating the folder manually"
+                .format(base_dir))
+
+logging.basicConfig(
+    format='%(asctime)s :%(name)s :%(levelname)s : %(message)s',
+    filename=log_file_dir, level=logging.INFO)
+console = logging.StreamHandler()
+console.setLevel(logging.INFO)
+logger = logging.getLogger('gensim.api')
+logger.addHandler(console)
+
+
+def download(dataset):
+    url = "https://github.com/chaitaliSaini/Corpus_and_models/releases/download/{f}/{f}.tar.gz".format(f=dataset)
+    data_folder_dir = os.path.join(base_dir, dataset)
+    data = catalogue(print_list=False)
+    corpuses = data['gensim']['corpus']
+    models = data['gensim']['model']
+    if dataset not in corpuses and dataset not in models:
+        logger.error(
```

Comment:
Raise an exception (not a logger.error + sys.exit)

Reply:
'corpuses' => 'corpora'